### ISSUE TITLE
status_notifier_get_type error: size of array

### ISSUE THREAD

**Issue by Geremia** [qTox/qTox#1521]
Upon compiling statusnotifier.c under Slackware, I get the following errors:

```
In file included from /usr/lib/glib-2.0/include/glibconfig.h:9:0,
                 from /usr/include/glib-2.0/glib/gtypes.h:34,
                 from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:25,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from src/platform/statusnotifier/statusnotifier.c:26:
src/platform/statusnotifier/statusnotifier.c: In function ‘status_notifier_get_type’:
/usr/include/glib-2.0/glib/gmacros.h:162:53: error: size of array ‘_GStaticAssertCompileTimeAssertion_0’ is negative
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                                     ^
/usr/include/glib-2.0/glib/gmacros.h:159:47: note: in definition of macro ‘G_PASTE_ARGS’
 #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
                                               ^
/usr/include/glib-2.0/glib/gmacros.h:162:44: note: in expansion of macro ‘G_PASTE’
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                            ^
/usr/include/glib-2.0/glib/gthread.h:250:5: note: in expansion of macro ‘G_STATIC_ASSERT’
     G_STATIC_ASSERT (sizeof *(location) == sizeof (gpointer));       \
     ^
/usr/include/glib-2.0/gobject/gtype.h:1503:7: note: in expansion of macro ‘g_once_init_enter’
   if (g_once_init_enter (&g_define_type_id__volatile))  \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1427:60: note: in expansion of macro ‘_G_DEFINE_TYPE_EXTENDED_BEGIN’
 #define G_DEFINE_TYPE_EXTENDED(TN, t_n, T_P, _f_, _C_)     _G_DEFINE_TYPE_EXTENDED_BEGIN (TN, t_n, T_P, _f_) {_C_;} _G_DEFINE_TYPE_EXTENDED_END()
                                                            ^
/usr/include/glib-2.0/gobject/gtype.h:1319:43: note: in expansion of macro ‘G_DEFINE_TYPE_EXTENDED’
 #define G_DEFINE_TYPE(TN, t_n, T_P)       G_DEFINE_TYPE_EXTENDED (TN, t_n, T_P, 0, {})
                                           ^
src/platform/statusnotifier/statusnotifier.c:171:1: note: in expansion of macro ‘G_DEFINE_TYPE’
 G_DEFINE_TYPE (StatusNotifier, status_notifier, G_TYPE_OBJECT)
 ^
In file included from /usr/include/glib-2.0/glib/gasyncqueue.h:34:0,
                 from /usr/include/glib-2.0/glib.h:34,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:25,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from src/platform/statusnotifier/statusnotifier.c:26:
/usr/include/glib-2.0/glib/gthread.h:251:17: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     (void) (0 ? (gpointer) *(location) : 0);                         \
                 ^
/usr/include/glib-2.0/gobject/gtype.h:1503:7: note: in expansion of macro ‘g_once_init_enter’
   if (g_once_init_enter (&g_define_type_id__volatile))  \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1427:60: note: in expansion of macro ‘_G_DEFINE_TYPE_EXTENDED_BEGIN’
 #define G_DEFINE_TYPE_EXTENDED(TN, t_n, T_P, _f_, _C_)     _G_DEFINE_TYPE_EXTENDED_BEGIN (TN, t_n, T_P, _f_) {_C_;} _G_DEFINE_TYPE_EXTENDED_END()
                                                            ^
/usr/include/glib-2.0/gobject/gtype.h:1319:43: note: in expansion of macro ‘G_DEFINE_TYPE_EXTENDED’
 #define G_DEFINE_TYPE(TN, t_n, T_P)       G_DEFINE_TYPE_EXTENDED (TN, t_n, T_P, 0, {})
                                           ^
src/platform/statusnotifier/statusnotifier.c:171:1: note: in expansion of macro ‘G_DEFINE_TYPE’
 G_DEFINE_TYPE (StatusNotifier, status_notifier, G_TYPE_OBJECT)
 ^
In file included from /usr/lib/glib-2.0/include/glibconfig.h:9:0,
                 from /usr/include/glib-2.0/glib/gtypes.h:34,
                 from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:25,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from src/platform/statusnotifier/statusnotifier.c:26:
/usr/include/glib-2.0/glib/gmacros.h:162:53: error: size of array ‘_GStaticAssertCompileTimeAssertion_1’ is negative
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                                     ^
/usr/include/glib-2.0/glib/gmacros.h:159:47: note: in definition of macro ‘G_PASTE_ARGS’
 #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
                                               ^
/usr/include/glib-2.0/glib/gmacros.h:162:44: note: in expansion of macro ‘G_PASTE’
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                            ^
/usr/include/glib-2.0/glib/gatomic.h:148:5: note: in expansion of macro ‘G_STATIC_ASSERT’
     G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
     ^
/usr/include/glib-2.0/glib/gthread.h:252:7: note: in expansion of macro ‘g_atomic_pointer_get’
     (!g_atomic_pointer_get (location) &&                             \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1503:7: note: in expansion of macro ‘g_once_init_enter’
   if (g_once_init_enter (&g_define_type_id__volatile))  \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1427:60: note: in expansion of macro ‘_G_DEFINE_TYPE_EXTENDED_BEGIN’
 #define G_DEFINE_TYPE_EXTENDED(TN, t_n, T_P, _f_, _C_)     _G_DEFINE_TYPE_EXTENDED_BEGIN (TN, t_n, T_P, _f_) {_C_;} _G_DEFINE_TYPE_EXTENDED_END()
                                                            ^
/usr/include/glib-2.0/gobject/gtype.h:1319:43: note: in expansion of macro ‘G_DEFINE_TYPE_EXTENDED’
 #define G_DEFINE_TYPE(TN, t_n, T_P)       G_DEFINE_TYPE_EXTENDED (TN, t_n, T_P, 0, {})
                                           ^
src/platform/statusnotifier/statusnotifier.c:171:1: note: in expansion of macro ‘G_DEFINE_TYPE’
 G_DEFINE_TYPE (StatusNotifier, status_notifier, G_TYPE_OBJECT)
 ^
In file included from /usr/include/glib-2.0/glib/gthread.h:34:0,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:34,
                 from /usr/include/glib-2.0/glib.h:34,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:25,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from src/platform/statusnotifier/statusnotifier.c:26:
/usr/include/glib-2.0/glib/gatomic.h:150:5: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     (gpointer) *(atomic);                                                    \
     ^
/usr/include/glib-2.0/glib/gthread.h:252:7: note: in expansion of macro ‘g_atomic_pointer_get’
     (!g_atomic_pointer_get (location) &&                             \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1503:7: note: in expansion of macro ‘g_once_init_enter’
   if (g_once_init_enter (&g_define_type_id__volatile))  \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1427:60: note: in expansion of macro ‘_G_DEFINE_TYPE_EXTENDED_BEGIN’
 #define G_DEFINE_TYPE_EXTENDED(TN, t_n, T_P, _f_, _C_)     _G_DEFINE_TYPE_EXTENDED_BEGIN (TN, t_n, T_P, _f_) {_C_;} _G_DEFINE_TYPE_EXTENDED_END()
                                                            ^
/usr/include/glib-2.0/gobject/gtype.h:1319:43: note: in expansion of macro ‘G_DEFINE_TYPE_EXTENDED’
 #define G_DEFINE_TYPE(TN, t_n, T_P)       G_DEFINE_TYPE_EXTENDED (TN, t_n, T_P, 0, {})
                                           ^
src/platform/statusnotifier/statusnotifier.c:171:1: note: in expansion of macro ‘G_DEFINE_TYPE’
 G_DEFINE_TYPE (StatusNotifier, status_notifier, G_TYPE_OBJECT)
 ^
In file included from /usr/lib/glib-2.0/include/glibconfig.h:9:0,
                 from /usr/include/glib-2.0/glib/gtypes.h:34,
                 from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:25,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from src/platform/statusnotifier/statusnotifier.c:26:
/usr/include/glib-2.0/glib/gmacros.h:162:53: error: size of array ‘_GStaticAssertCompileTimeAssertion_2’ is negative
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                                     ^
/usr/include/glib-2.0/glib/gmacros.h:159:47: note: in definition of macro ‘G_PASTE_ARGS’
 #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
                                               ^
/usr/include/glib-2.0/glib/gmacros.h:162:44: note: in expansion of macro ‘G_PASTE’
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                            ^
/usr/include/glib-2.0/glib/gthread.h:257:5: note: in expansion of macro ‘G_STATIC_ASSERT’
     G_STATIC_ASSERT (sizeof *(location) == sizeof (gpointer));       \
     ^
/usr/include/glib-2.0/gobject/gtype.h:1517:7: note: in expansion of macro ‘g_once_init_leave’
       g_once_init_leave (&g_define_type_id__volatile, g_define_type_id); \
       ^
/usr/include/glib-2.0/gobject/gtype.h:1427:117: note: in expansion of macro ‘_G_DEFINE_TYPE_EXTENDED_END’
 #define G_DEFINE_TYPE_EXTENDED(TN, t_n, T_P, _f_, _C_)     _G_DEFINE_TYPE_EXTENDED_BEGIN (TN, t_n, T_P, _f_) {_C_;} _G_DEFINE_TYPE_EXTENDED_END()
                                                                                                                     ^
/usr/include/glib-2.0/gobject/gtype.h:1319:43: note: in expansion of macro ‘G_DEFINE_TYPE_EXTENDED’
 #define G_DEFINE_TYPE(TN, t_n, T_P)       G_DEFINE_TYPE_EXTENDED (TN, t_n, T_P, 0, {})
                                           ^
src/platform/statusnotifier/statusnotifier.c:171:1: note: in expansion of macro ‘G_DEFINE_TYPE’
 G_DEFINE_TYPE (StatusNotifier, status_notifier, G_TYPE_OBJECT)
 ^
make: *** [statusnotifier.o] Error 1
```

**Comment by Geremia** [comment]
The issue was that I included 32 bit glib2 in the `INCLUDEPATH += "etc."` of my *.pro file. I deleted those and retained only the 64 bit library paths, and it compiled without the above error.

**Comment by aagmangour** [comment]
Hi ,
I am compiling glib  for ARMv8A and I am facing the same issue .
Could you please guide me where is this .pro  file present or where should I change the 32 to 64.
Thanks in Advance

**Comment by WileyECoyote** [comment]
argument to g_once_init_enter is the wrong size for the target. This error will occur if the address of an "unsigned int" is passed for a 64 bit build,  which would be okay for a 32 bit build but should be an unsigned long for 64 bit. If defined using GType  then will be the size glib is building, maybe pass -m64 / -m32 on the command line to control the build target

**Comment by JackZ0** [comment]
In file included from /usr/lib/arm-linux-gnueabihf/glib-2.0/include/glibconfig.h:9:0,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from gst_private.h:36,
                 from gstelementfactory.c:60:
gstelementfactory.c: In function 'gst_element_factory_create':
/usr/include/glib-2.0/glib/gmacros.h:181:53: error: size of array '_GStaticAssertCompileTimeAssertion_3' is negative
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                                     ^
/usr/include/glib-2.0/glib/gmacros.h:178:47: note: in definition of macro 'G_PASTE_ARGS'
 #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
                                               ^
/usr/include/glib-2.0/glib/gmacros.h:181:44: note: in expansion of macro 'G_PASTE'
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                            ^
/usr/include/glib-2.0/glib/gatomic.h:159:5: note: in expansion of macro 'G_STATIC_ASSERT'
     G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
     ^
gstelementfactory.c:393:8: note: in expansion of macro 'g_atomic_pointer_compare_and_exchange'
   if (!g_atomic_pointer_compare_and_exchange (
        ^
gstelementfactory.c:393:3: error: invalid use of void expression
   if (!g_atomic_pointer_compare_and_exchange (
   ^
gstelementfactory.c:393:3: error: incompatible type for argument 1 of '__sync_bool_compare_and_swap'
make[4]: *** [libgstreamer_0.10_la-gstelementfactory.lo] Error 1
make[4]: Leaving directory `/root/gstreamer-0.10.32/gst'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/root/gstreamer-0.10.32/gst'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/root/gstreamer-0.10.32/gst'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/root/gstreamer-0.10.32'
make: *** [all] Error 2
